persist/fs/commitlog/chunk_reader.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package commitlog
+
+import (
+	"bufio"
+	"os"
+
+	"github.com/m3db/m3db/digest"
+)
+
+type chunkReader struct {
+	fd        *os.File
+	buffer    *bufio.Reader
+	remaining int
+	charBuff  []byte
+}
+
+func newChunkReader(bufferLen int) *chunkReader {
+	return &chunkReader{
+		buffer:   bufio.NewReaderSize(nil, bufferLen),
+		charBuff: make([]byte, 1),
+	}
+}
+
+func (r *chunkReader) reset(fd *os.File) {
+	r.fd = fd
+	r.buffer.Reset(fd)
+	r.remaining = 0
+}
+
+func (r *chunkReader) readHeader() error {
+	header, err := r.buffer.Peek(chunkHeaderLen)
+	if err != nil {
+		return err
+	}
+
+	sizeStart, sizeEnd :=
+		0, chunkHeaderSizeLen
+	checksumSizeStart, checksumSizeEnd :=
+		sizeEnd, sizeEnd+chunkHeaderSizeLen
+	checksumDataStart, checksumDataEnd :=
+		checksumSizeEnd, checksumSizeEnd+chunkHeaderChecksumDataLen
```

Comment:
Perhaps move these constants to the top of this file? i.e. `chunkHeaderSizeLen` and `chunkHeaderChecksumDataLen`